During COVID-19, a lot of Americans took up baking as a hobby. Facebook, Instagram, and Twitter were all flooded with pictures of baked bread, croissants, cookies, and cakes. This led to the "Quarantine 15" meme where people claimed they gained 15 lbs from the excess of lockdown cooking and carbs. 

We are wondering whether the United States surge in COVID-19 had any affect on the sale of dairy products during this time, especially since so many people took up baking as a hobby. We are using the USDA Dairy Glance dataset, which includes various dairy product statistics from January 2020 to March 2021.

Our first hypothesis is as follows: Did the surge in COVID-19 cases lead to an  increase in butter output? Butter is a staple for many baked goods, so we hypothesize that this baking craze could have led to an increase in butter output from farmers. 

Here, we apply Pearson's correlation to see if butter output and cumulative US covid cases are correlated.

Ho: Butter output from January to June and cumulative monthly USA covid cases from January to June are not correlated

H1: Butter output from January to June and cumulative monthly USA covid cases from January to June are correlated

In [128]:
import pandas as pd
import numpy as np

In [135]:
butter = pd.read_csv('butter.csv')
file = 'https://raw.githubusercontent.com/michaelofsbu/CSE-544-Datasets/main/US-all/US_confirmed.csv'
us = pd.read_csv(file)

In [136]:
us = us.T.drop("State")

In [137]:
us.index = pd.to_datetime(us.index)

In [138]:
us['cumulative'] = us.sum(axis=1)

In [166]:
us['cumulative'] = us['cumulative'].diff()

In [169]:
us['cumulative'][0] = 8

In [170]:
us = us[["cumulative"]]

In [171]:
us = us[(us.index >= '2020-01-22') & (us.index <= '2020-06-30')]

In [172]:
us

cumulative
2020-01-31         8.0
2020-02-28        15.0
2020-03-31    186727.0
2020-04-30    876564.0
2020-05-31    715946.0
2020-06-30    839557.0

In [173]:
us = us[(us.index == '2020-01-31') | (us.index == '2020-02-28') | (us.index == '2020-03-31') | (us.index == '2020-04-30') | (us.index == '2020-05-31') | (us.index == '2020-06-30')]

In [174]:
butter = butter[0:6]

In [175]:
butter_xbar = butter['Butter Output'].mean()

In [176]:
cases_ybar = us['cumulative'].mean()

In [177]:
butter_output = np.array(butter['Butter Output'])
cases_cumulative = np.array(us['cumulative'])

In [178]:
numerator = 0
x_sum_denominator = 0
y_sum_denominator = 0
for i in range(len(butter_output)):
  x = butter_output[i] - butter_xbar
  y = cases_cumulative[i] - cases_ybar
  numerator += x * y

  x_sum_denominator += x**2
  y_sum_denominator += y**2

print(abs(numerator / np.sqrt(x_sum_denominator * y_sum_denominator)))

0.20102796503194495


Baed off of the Pearson correlation score, since the value is 0.201 and |0.201| is < 0.5, we accept the null hypothesis that butter output from January to June and Cumulative USA covid cases from January to June are not correlated.